doc/adrs/107_improve_frontend_selection_support.adoc

```diff
@@ -0,0 +1,75 @@
+= ADR-107 - Improve frontend selection support
+
+== Context
+
+Selection handling is currently managed by the `Workbench` frontend component, which is only part (although the main one) of the project edition view/page (EditProjectView).
+Actions of the EditProjectView which are not part of the workbench (for example in the navbar and its menu) should also be able to read and set the selection.
+
+== Decision
+
+We will extract selection handling from `Workbench(Machine).tsx` into a custom hook.
+This will simplify the Workbench state machine.
+
+It will also isolate selection handling implementation and enable a lot of simplications where we used to drill down `selection` and/or `setSelection` using props.
+Components which need to read/set the selection will be able to to so using:
+
+```js
+const {selection, setSelection} = useSelection();
+```
+
+Some components (notably the worbench itself) need to be able to react to selection changes.
+They can do this with a `useEffect()` which reacts to changes in the selection value:
+
+```js
+const {selection} = useSelection();
+useEffect(() => {
+  /* handle the new selection value */
+}, [selection]);
+```
+
+Components which used to get the selection and/or setSelection callback as part of their props just to pass them to their children will be simplified.
+
+The state needed for the new hook will be handle by the `SelectionContext` and it will be possible to initialize the selection.
+
+```js
+
+export const SelectionContext = React.createContext(undefined);
+
+export const SelectionContextProvider = ({ initialSelection, children }) => {
+  const [state, setState] = useState({
+    selection: initialSelection ?? { entries: [] },
+  });
+
+  const setSelection = useCallback((selection: Selection) => {
+    setState((prevState) => ({ ...prevState, selection }));
+  }, []);
+
+  return (
+    <SelectionContext.Provider
+      value={{ selection: state.selection, setSelection }}>
+      {children}
+    </SelectionContext.Provider>
+  );
+}
+
+```
+
+The `SelectionContextProvider` will be used as the parent of the `Workbench` and the `Project Navbar` in `EditProjectView`:
+
+```js
+
+return (
+  <SelectionContextProvider initialSelection={initialSelection}>
+    {navbar}
+    {main}
+    <Snackbar />
+  </SelectionContextProvider>
+);
+
+```
+
+This makes sense because the selection is relative to a particular project/editing context.
```

Comment:
Actually, maybe not always.
I am working on Portals support, where one I need more control on how the selection *inside* the representations displayed in a portal are synchronized with the global, project-wide selection.

This is just a remark, [it seems](https://react.dev/reference/react/useContext#overriding-context-for-a-part-of-the-tree) that even with this new approach I should be able to locally override the selection handling for my case. It might be worth mentioning, you decide.